mirrulations-client/src/mirrclient/client.py

```diff
@@ -236,30 +237,41 @@ def perform_attachment_job(self, url, job_id):
         -------
         a dict of encoded files
         """
-        non_api_url = url
         url = url + f'?api_key={self.api_key}'
-        response_from_related = requests.get(url, timeout=10).json()
+        response_json = requests.get(url, timeout=10).json()
 
-        # Get attachments
-        try:
-            file_info = \
-                response_from_related["data"][0]["attributes"]["fileFormats"]
-            if not file_info:
-                raise TypeError
-            file_urls, file_types = get_urls_and_formats(file_info)
-        except IndexError:
-            # if related attachments link is an empty data =[] json
-            print(f'FAILURE: Empty attachment list from {non_api_url}')
-            return {}
-        except KeyError as error:
-            print(f"FAILURE: {error} from {non_api_url}")
+        if not self.does_attachment_exists(response_json):
+            print(f"No attachments to download from {url}")
             return {}
-        except TypeError as error:
-            print(f"FAILURE: {error} from {non_api_url}")
-            return {}
-        print(f'Performing attachment job {job_id}')
+
+        # Get Attachments
+        print(f"SUCCESS: Performing attachment job {job_id}")
+        file_info = \
+            response_json["data"][0]["attributes"]["fileFormats"]
+        file_urls, file_types = get_urls_and_formats(file_info)
         return self.download_attachments(file_urls, file_types, job_id)
 
+    def does_attachment_exists(self, attachment_json):
+        """
+        Validates whether a json for an attachment is valid to continue
+        download process. Invalid JSON means no attachment(s) available
+
+        RETURNS
+        -------
+        True or False depending if there is an attachment available to download
+        """
+        # handle keyError
+        if "data" not in attachment_json:
+            return False
+        # Handles IndexError
+        if attachment_json.get("data") == []:
+            return False
+        # Handles NoneType
+        if len(attachment_json['data']) >= 1:
+            if not attachment_json['data'][0]['attributes']['fileFormats']:
```

Comment:
It would be best if we also checked for the existence of `attributes` and `fileFormats` before attempting to retrieve them from the JSON.